src/profile-logic/process-profile.js

```diff
@@ -754,6 +755,43 @@ function _processThread(
     stringTable,
     samples,
   };
+
+  function processJsTracer() {
+    // Optionally extract the JS Tracer information, if they exist.
+    const { jsTracerEvents } = thread;
+    const { jsTracerDictionary } = processProfile;
+    if (jsTracerEvents && jsTracerDictionary) {
+      // Add the JS tracer's strings to the thread's existing string table, and create
+      // a mapping from the old string indexes to the new ones. Use an Array rather
+      // than a Map because it saves ~150ms out of ~300ms in one example.
+      const geckoToProcessedStringIndex: number[] = new Array(
+        jsTracerDictionary.length
+      );
+      for (let i = 0; i < jsTracerDictionary.length; i++) {
+        geckoToProcessedStringIndex[i] = newThread.stringTable.indexForString(
+          jsTracerDictionary[i]
```

Comment:
This is probably better suited in a followup issue, but it would be nice to see the event names as "script:lineno:colno" in the stack based view.  Or alternatively, a tooltip over the event will tell you which "script:lineno:colno" event it belongs to.   Currently it is not possible to tell which function any of the events are operating on.

Reply:
Filed here: https://github.com/devtools-html/perf.html/issues/1504